pkg/bootstrap/bootstrap_config.go

```diff
@@ -229,6 +230,12 @@ func WriteBootstrap(config *meshconfig.ProxyConfig, node string, epoch int, pilo
 	opts["cluster"] = config.ServiceCluster
 	opts["nodeID"] = node
 
+	// Get the platform zone if available.
+	z := platform.GetZone()
+	if z != "" {
+		opts["zone"] = z
+	}
```

Comment:
please dont set this Zone. Set the locality field in the node metadata instead. Locality has multiple pieces (region/zone/sub-zone).

Reply:
As per the [Locality's documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/core/base.proto#core-locality), the zone field maps to the GCP zone/AWS availability zone. In this context, zone can be consumed without region/sub-zone. Why wouldn't we want to use this field?

Reply:
because Pilot wont respect it :) properly. We have switched to using the locality field instead of zone cli flag

Reply:
Sorry I might be missing something, but this isn't a zone cli flag. This value populates `Node.Locality.Zone` in the bootstrap template.

Are you saying that Pilot ignores the field `Node.Locality` and instead looks inside `Node.Metadata` for a key corresponding to the real locality?

The motivation for this change is that we use `Locality.Zone` in our managed version of Pilot. Pilot could ignore this field if it has no use for it, but I imagine this could be useful for Pilot as well (at the least for visibility/debug).

Reply:
oh my bad.. in that case, you need to set the region as well. If you dont set region but simply set the zone alone, then pilot will treat it as though no locality is set.

Reply:
Sounds good, refactored the code to set region and zone.